controllers/utils/utils.go

```diff
@@ -165,6 +170,45 @@ func VerifyIfCapabilityIsEnabled(ctx context.Context, cli client.Client, capabil
 
 }
 
+// VerifyIfMeshAuthorizationIsEnabled func checks if Authorization has been configured for
+// ODH platform. That check is done by verifying the presence of an Istio AuthorizationPolicy
+// resource. This resource is expected to be present for both the ODH managed and unmanaged
+// setups of the authorization capability. If it is not present, it is inferred that the
+// authorization capability is not available.
+func VerifyIfMeshAuthorizationIsEnabled(ctx context.Context, cli client.Client) (bool, error) {
+	// TODO: The VerifyIfCapabilityIsEnabled func was supposed to be a generic way to check
```

Comment:
@bartoszmajsak I agree with all you wrote. In the long run we should remove this, and I'm hoping we can remove the workaround ASAP.

However, for that, I think we now are on the need of a _feature refinement_ around managed/unmanaged deps. As it is now, "managed" is not really managed, but just "setup and forget". So, pretty much like the operator just doing the setup, then leaving the stack as unmanaged (despite the `managementStatus: Managed`).

Given current situation, I think that whatever mechanism we do for figuring out what are the capabilities for Unmanged, such mechanism still applies to Managed. We need to sit and write down if this is what we really want, and make the operator consistent with this -  as I think this would also impact on ServiceMesh capability detection.

So, I simply wanted to have this sorted out, given the release is due in a few days. This was the easiest I could think of. But I do agree we should work in the long term solution.

BTW...

> Can we, in case of unmanaged, instruct the user to share this information somehow?

I'm seeing this like adding the `managementState` field under `auth` of the DSCI :confused: